src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -492,6 +492,25 @@ parseNumberBinder = NumberBinder <$> (sign <*> number)
          <|> (symbol' "+" >> return id)
          <|> return id
 
+parseTypedBinder :: TokenParser Binder
+parseTypedBinder = buildPostfixParser postfixTable typeableBinder P.<?> "typed binder"
+    where
+      postfixTable = [ \b -> flip TypedBinder b <$> (P.try (indented *> doubleColon) *> parseType)
+                     ]
+      typeableBinder = P.choice (map P.try
```

Comment:
Why not just allow any binder here? Also, I think you can move the call to `buildPostfixParser` into `parseBinder`, like the `Type` parser does. That way, all the "operators" are in one place.